From 9346dc7aa8f7b8c0c5f60770a84eb2bd11f49558 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Tue, 31 Jul 2012 00:06:40 +0800
Subject: fix crash issue while playing certain streaming media

BZ:49305

In the play response of certain streaming media, the url just contains part of the full url.
Change the url matching method to enable this case rather than crash as before.

orignal-patch
30070:libstagefright: patch porting from R2 to R3 for RTSP
16854[R1]:fix crash issue while playing certain streaming media(BZ:7540)

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: Ie6c64f0588bed36f88a3aa89c4cbea38251ac000
Orig-Change-Id: I69b4e74d415cdf006ae87a7e6138264af2603771
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/rtsp/MyHandler.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/rtsp/MyHandler.h b/media/libstagefright/rtsp/MyHandler.h
index ccb6788..0f7879b 100644
--- a/media/libstagefright/rtsp/MyHandler.h
+++ b/media/libstagefright/rtsp/MyHandler.h
@@ -1413,7 +1413,7 @@ struct MyHandler : public AHandler {
 
             size_t trackIndex = 0;
             while (trackIndex < mTracks.size()
-                    && !(val == mTracks.editItemAt(trackIndex).mURL)) {
+                    && (-1 == mTracks.editItemAt(trackIndex).mURL.find(val.c_str(), 0))) {
                 ++trackIndex;
             }
             CHECK_LT(trackIndex, mTracks.size());
-- 
1.7.4.1

